backend/coreapp/middleware.py

```diff
@@ -34,6 +34,11 @@ def set_user_profile(get_response):
     """
 
     def middleware(request: Request):
+        # Skip if the request is from SSR
+        if "User-Agent" in request.headers and "node-fetch" in request.headers["User-Agent"]:
```

Comment:
I can see this potentially causing issues in the future wrt node-fetch-based API consumers, but we don't have any so they can complain later.

Reply:
Perhaps we could add a custom `X-No-Cookie: yes` header to all SSR API requests and detect that way?

Reply:
yeah that's fine, I just didn't know how to modify the header but if you help me figure it out, I agree that's better